tests/test_vpc.py

```diff
@@ -559,6 +562,133 @@ def test_interface_subnet(self):
         self.assertEqual([g['GroupId'] for g in results[0]['Groups']], [qsg_id])
 
 
+class NetworkAddrTest(BaseTest):
+
+    @staticmethod
+    def release_if_still_present(ec2, network_address):
+        # @todo (2018-04-23 jonathana) Throw some logging in here just to see when the cleanup
+        # we're doing from self.addCleanup actually fires
+        try:
+            release_spec = AddressRelease.gen_assoc_spec(network_address)
+            ec2.release_address(**release_spec)
+        except BotoClientError as e:
+            # Swallow the condition that the elastic ip wasn't there (meaning the test should have deleted it),
+            # re-raise any other boto client error
+            if not(e.response['Error']['Code'] == 'InvalidAllocationID.NotFound' and network_address['AllocationId'] in e.response['Error']['Message']):
+                raise e
+
+    # This doesn't really need to be an instance method, it could easily be
+    # classmethod, except we want to use .assertEquals, which is an instance
+    # method
+    def create_ec2_instance(self, ec2, size='t2.nano', ami_id='ami-43a15f3e'):
+        result = ec2.run_instances(ImageId=ami_id,
+                                   InstanceType=size,
+                                   MaxCount=1,
+                                   MinCount=1,
+                                   DisableApiTermination=False,
+                                   )
+        self.assertEquals(len(result['Instances']), 1)
+        self.addCleanup(ec2.terminate_instances, InstanceIds=[i['InstanceId'] for i in result['Instances']])
+
+        instance_data = result['Instances'][0]
+        while instance_data['State']['Code'] != 16:
+            time.sleep(5)
+            result = ec2.describe_instances(InstanceIds=(instance_data['InstanceId'],))
+            self.assertEquals(len(result['Reservations'][0]['Instances']), 1)
+            instance_data = result['Reservations'][0]['Instances'][0]
+
+        return instance_data
+
+    def allocate_network_address(self, ec2, domain_type):
+        network_addr = ec2.allocate_address(Domain=domain_type)
+        self.addCleanup(self.release_if_still_present, ec2, network_addr)
+        return network_addr
+
+    def core_release_op(self, factory, ec2, network_addr, force=False):
```

Comment:
just a naming sugestion, would recommend renaming to assert_policy_release() .. core release op is a big ambigious.